lib/parser/models.ex

```diff
@@ -0,0 +1,479 @@
+defmodule Thrift.Parser do
+
+  defmodule Types do
+    defmodule Primitive do
+      @type t :: :bool | :i8 | :i16 | :i64 | :binary | :double | :byte | :string
+    end
+
+    defmodule Ident do
+      @type t :: String.t
+    end
+
+    defmodule Standalone do
+      @type t :: Ident.t | Primitive.t
+    end
+
+    defmodule List do
+      @type t :: {:list, Types.t}
+    end
+
+    defmodule Map do
+      @type t  :: {:map, {Types.t, Types.t}}
+    end
+
+    defmodule Set do
+      @type t :: {:set, Types.t}
+    end
+
+    defmodule Container do
+      @type t :: List.t | Map.t | Set.t
+    end
+
+    @type t :: Container.t | Standalone.t
+  end
+
+  defmodule Literals do
+    defmodule Primitive do
+      @type t :: integer | boolean | String.t | float
+    end
+
+    defmodule List do
+      @type t :: [Literals.t]
+    end
+
+    defmodule Map do
+      @type t :: %{Literals.t => Literals.t}
+    end
+
+    defmodule Container do
+      @type t :: Map.t | List.t
+    end
+
+    @type t :: Container.t | Primitive.t
+    @type s :: atom
+  end
+
+
+  defmodule Shell do
+    @yellow IO.ANSI.yellow
+    @red IO.ANSI.red
+    @reset IO.ANSI.reset
+
+    def warn(msg) do
+      IO.puts "#{@yellow}#{msg}#{@reset}"
+    end
+
+    def error(msg) do
+      IO.puts "#{@red}#{msg}#{@reset}"
+    end
+  end
+
+  defmodule Conversions do
+    def atomify(nil), do: nil
+    def atomify(l) when is_list(l) do
+      List.to_atom(l)
+    end
+
+    def stringify(l) when is_list(l) do
+      List.to_string(l)
+    end
+
+    def cast(_, nil) do
+      nil
+    end
+
+    def cast(:int, val) do
+      val
+    end
+
+    def cast(:double, val) do
+      val
+    end
+
+    def cast(:string, val) do
+      List.to_string(val)
+    end
+
+    def cast({:set, type}, val) do
+      MapSet.new(val, &cast(type, &1))
+    end
+
+    def cast({:map, {key_type, val_type}}, val) do
+      Enum.into(val, %{}, fn {k, v} ->
+        {cast(key_type, k), cast(val_type, v)}
+      end)
+    end
+
+    def cast(_, val) do
+      val
+    end
+  end
+
+  defmodule Models do
+    alias Thrift.Parser.Types
+    alias Thrift.Parser.Literals
+
+    defmodule Namespace do
+      @moduledoc """
+      A Thrift namespace.
+      The namespace is a language-specific place where the generated structs are
+      placed.
+      """
+      defstruct name: nil, path: nil
+      import Thrift.Parser.Conversions
+
+      @spec new(char_list, char_list) :: %Namespace{}
+      def new(name, path) do
+        %Namespace{name: atomify(name), path: stringify(path)}
```

Comment:
I think this now reenforces my earlier thought that it would be nicer to just use `List.to_atom` and `List.to_string` directly. I find those explicit conversions easier to reason about.

That would let us drop `stringify` entirely because it's only used here an in `Include.new/1`.